plugins/inputs/pf/pf.go

```diff
@@ -0,0 +1,207 @@
+// +build freebsd
+
+package pf
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const measurement = "pf"
+const pfctlCommand = "pfctl"
+
+type PF struct {
+	UseSudo    bool
+	StateTable StateTable
+	infoFunc   func() (string, error)
+}
+
+func (pf *PF) Description() string {
+	return "Gather counters from PF"
+}
+
+func (pf *PF) SampleConfig() string {
+	return `
+  ## PF require root access on most systems.
+  ## Setting 'use_sudo' to true will make use of sudo to run pfctl.
+  ## Users must configure sudo to allow telegraf user to run pfctl with no password.
+  ## pfctl can be restricted to only list command "pfctl -s info".
+  use_sudo = false
+`
+}
+
+// Gather is the entrypoint for the plugin.
+func (pf *PF) Gather(acc telegraf.Accumulator) error {
+	o, err := pf.infoFunc()
+	if err != nil {
+		acc.AddError(err)
+		return nil
+	}
+
+	if perr := pf.parsePfctlOutput(o, acc); perr != nil {
+		acc.AddError(perr)
+	}
+	return nil
+}
+
+var errParseHeader = fmt.Errorf("Cannot find header in %s output", pfctlCommand)
+
+func errMissingData(tag string) error {
+	return fmt.Errorf("struct data for tag \"%s\" not found in %s output", tag, pfctlCommand)
+}
+
+var stateTableHeaderRE = regexp.MustCompile("^State Table")
+var anyTableHeaderRE = regexp.MustCompile("^[A-Z]")
+
+func (pf *PF) parsePfctlOutput(pfoutput string, acc telegraf.Accumulator) error {
+	lines := strings.Split(pfoutput, "\n")
```

Comment:
I recommend using a bufio.Scanner to deal with non-unix line endings.  Might make it easier to pass the scanner into child functions too instead of using a index range.